src/NuGet.Core/NuGet.Packaging/Signing/Verification/NuGetSignatureWhitelistObject.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Packaging.Signing
+{
+    public class NuGetSignatureWhitelistObject
```

Comment:
Consider the following - 

```
    public class NuGetSignatureWhitelistObject
    {
        public string Value { get; }

        public NuGetSignatureWhitelistObjectType Type { get; }

        public NuGetSignatureWhitelistObject(NuGetSignatureWhitelistObjectType type, string value)
        {
            Type = type;
            Value = value;
        }

        public NuGetSignatureWhitelistObject(string value)
        {
            Type = NuGetSignatureWhitelistObjectType.Any;
            CertificateFingerprint = fingerprint;
        }
    }
```

And - 

```
    public enum NuGetSignatureWhitelistObjectType
    {
        PrimarySignatureCertificateHash     = 0,
        RepositorySignatureCertificateHash = 1
    }
```

Just like we discussed offline, this makes the implementation open to new whitelist objects.